RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -227,44 +368,90 @@ EvtPlaneProducer::EvtPlaneProducer(const edm::ParameterSet &iConfig)
       minvtx_(iConfig.getParameter<double>("minvtx")),
       maxvtx_(iConfig.getParameter<double>("maxvtx")),
-      dzerr_(iConfig.getParameter<double>("dzerr")),
+      flatnvtxbins_(iConfig.getParameter<int>("flatnvtxbins")),
+      flatminvtx_(iConfig.getParameter<double>("flatminvtx")),
+      flatdelvtx_(iConfig.getParameter<double>("flatdelvtx")),
+      dzdzerror_(iConfig.getParameter<double>("dzdzerror")),
+      d0d0error_(iConfig.getParameter<double>("d0d0error")),
+      pterror_(iConfig.getParameter<double>("pterror")),
+      chi2perlayer_(iConfig.getParameter<double>("chi2perlayer")),
+      dzdzerror_pix_(iConfig.getParameter<double>("dzdzerror_pix")),
       chi2_(iConfig.getParameter<double>("chi2")),
+      nhitsValid_(iConfig.getParameter<int>("nhitsValid")),
       FlatOrder_(iConfig.getParameter<int>("FlatOrder")),
       NumFlatBins_(iConfig.getParameter<int>("NumFlatBins")),
       caloCentRef_(iConfig.getParameter<double>("caloCentRef")),
       caloCentRefWidth_(iConfig.getParameter<double>("caloCentRefWidth")),
-      CentBinCompression_(iConfig.getParameter<int>("CentBinCompression")) {
+      CentBinCompression_(iConfig.getParameter<int>("CentBinCompression")),
+      cutEra_(iConfig.getParameter<int>("cutEra"))
+
+{
+  switch (cutEra_) {
+    case 0:
+      cuts = new EPCuts(
+          EP_ERA::ppReco, pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
+      break;
+    case 1:
+      cuts = new EPCuts(
+          EP_ERA::HIReco, pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
+      break;
+    case 2:
+      cuts = new EPCuts(
+          EP_ERA::Pixel, pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
+      break;
+    case 3:
+      cuts = new EPCuts(
+          EP_ERA::GenMC, pterror_, dzdzerror_, d0d0error_, chi2perlayer_, dzdzerror_pix_, chi2_, nhitsValid_);
+      break;
+    default:
+      cuts = nullptr;
+  }
   nCentBins_ = 200.;
 
   if (iConfig.exists("nonDefaultGlauberModel")) {
     centralityMC_ = iConfig.getParameter<std::string>("nonDefaultGlauberModel");
   }
   centralityLabel_ = centralityVariable_ + centralityMC_;
 
-  centralityBinToken = consumes<int>(centralityBinTag_);
-
-  vertexToken = consumes<std::vector<reco::Vertex>>(vertexTag_);
+  centralityBinToken_ = consumes<int>(centralityBinTag_);
 
-  caloToken = consumes<CaloTowerCollection>(caloTag_);
+  vertexToken_ = consumes<std::vector<reco::Vertex>>(vertexTag_);
 
-  castorToken = consumes<std::vector<reco::CastorTower>>(castorTag_);
+  bStrack_packedPFCandidates_ = (trackTag_.label().find("packedPFCandidates") != std::string::npos);
+  bScalo_particleFlow_ = (caloTag_.label().find("particleFlow") != std::string::npos);
+  if (bStrack_packedPFCandidates_) {
+    packedToken_ = consumes<edm::View<pat::PackedCandidate>>(trackTag_);
+    lostToken_ = consumes<edm::View<pat::PackedCandidate>>(losttrackTag_);
```

Comment:
why a view? It has some overhead and is needed if polymorphic access is required
```suggestion
    packedToken_ = consumes<pat::PackedCandidateCollection>(trackTag_);
    lostToken_ = consumes<pat::PackedCandidateCollection>(losttrackTag_);
```